validation/superchain-version_test.go

```diff
@@ -0,0 +1,122 @@
+		888:          true,
+		957:          true,
+		997:          true,
+		8453:         true,
+		34443:        true,
+		58008:        true,
+		84531:        true,
+		84532:        true,
+		7777777:      true,
+		11155420:     true,
+		11763071:     true,
+		999999999:    true,
+		129831238013: true,
+	}
+
+	isSemverAcceptable := func(desired, actual string) bool {
+		return desired == actual
+	}
+
+	checkOPChainSatisfiesSemver := func(chain *ChainConfig) {
+		rpcEndpoint := Superchains[chain.Superchain].Config.L1.PublicRPC
+
+		require.NotEmpty(t, rpcEndpoint)
+
+		client, err := ethclient.Dial(rpcEndpoint)
+		require.NoErrorf(t, err, "could not dial rpc endpoint %s", rpcEndpoint)
+
+		contractNames := []string{
+			"L1CrossDomainMessenger",
+			"L1ERC721Bridge",
+			"L1StandardBridge",
+			"L2OutputOrcale",
+			"OptimismMintableERC20Factory",
+			"OptimismPortal",
+			"SystemConfig",
+		}
+
+		for _, contractName := range contractNames {
+			desiredSemver, err := SuperchainSemver[chain.Superchain].VersionFor(contractName)
+			require.NoError(t, err)
+
+			// ASSUMPTION: we will check the version of the implementation via the declared proxy contract
+			proxyContractName := contractName + "Proxy"
+
+			contractAddress, err := Addresses[chain.ChainID].AddressFor(proxyContractName)
+			require.NoErrorf(t, err, "%s/%s.%s.version= UNSPECIFIED (desired version %s)", chain.Superchain, chain.Name, proxyContractName, desiredSemver)
+
+			actualSemver, err := getVersionWithRetries(context.Background(), common.Address(contractAddress), client)
+			require.NoErrorf(t, err, "RPC endpoint %s: %s", rpcEndpoint)
+
+			require.Condition(t, func() bool { return isSemverAcceptable(desiredSemver, actualSemver) },
+				"%s/%s.%s.version=%s (UNACCEPTABLE desired version %s)", chain.Superchain, chain.Name, proxyContractName, actualSemver, desiredSemver)
+
+			t.Logf("%s/%s.%s.version=%s (acceptable compared to %s)", chain.Superchain, chain.Name, proxyContractName, actualSemver, desiredSemver)
+
+		}
+	}
+
+	for chainID, chain := range OPChains {
+		if !isExcluded[chainID] {
+			checkOPChainSatisfiesSemver(chain)
+		}
+	}
+}
+
+// getVersion will get the version of a contract at a given address, if it exposes a version() method.
+func getVersion(ctx context.Context, addr common.Address, client *ethclient.Client) (string, error) {
+	isemver, err := bindings.NewISemver(addr, client)
+	if err != nil {
+		return "", fmt.Errorf("%s: %w", addr, err)
+	}
+	version, err := isemver.Version(&bind.CallOpts{
+		Context: ctx,
+	})
+	if err != nil {
+		return "", fmt.Errorf("%s: %w", addr, err)
+	}
+
+	return version, nil
+
+}
+
+// getVersionWithRetries is a wrapper for getVersion
+// which retries up to 10 times with exponential backoff.
+func getVersionWithRetries(ctx context.Context, addr common.Address, client *ethclient.Client) (string, error) {
+	const maxAttempts = 10
+	return retry.Do(context.Background(), maxAttempts, retry.Exponential(), func() (string, error) {
```

Comment:
[nit] Pass in `ctx` from above to allow this to be cancelled with the parent context

Reply:
[94b84e7](https://github.com/ethereum-optimism/superchain-registry/pull/62/commits/94b84e74fc6d0516da5214369cb48631212da86a)